### PR TITLE
View an approved post's details by clickings its replies chip.

### PR DESCRIPTION
**Pull request by vfcheung**
To test, got to the Approved Posts tab on the forum or `/forum/approved` and click on the replies chip.

Note: in `ForumApprovedPostDetails.js`, there are three database fetches. I do not know how to best handle this without repeating code.

### CODE REVIEW

File: src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js
Comment (enmingliu) [inline src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js:59]: About the 3 repeated database fetches, I can see that the data necessary for the approved post details in the users (display name) and category (category) collections are already available in approved posts list. Maybe we should be passing this data from the approved posts list to the approved post details, but I'm not sure if that's possible from the way you set up routing.

File: src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js
Comment (enmingliu) [inline src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js:34]: Is making this function a promise with async necessary here? Do we need to await anything before we set the post hook?

File: src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js
Comment (enmingliu) [inline src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js:88]: Also, just a note, if we do end up not having to make database fetches for the user display name, then I'm not sure if its necessary for the remaining database fetches to be in `ForumApprovedPostDetails` when it could be in `ForumPostDetailsList`

File: src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js
Comment (enmingliu) [inline src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js:14]: Does the width have to be a constant value? Can it be relative?

File: src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js
Comment (enmingliu) [inline src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js:42]: Again, we use the same icons in the approved post list & the approved post details, it'd be great if we could pass data like this action buttons array from the post list to the post details.

**Review by enmingliu — CHANGES_REQUESTED** [review]

File: src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js
Comment (vfcheung) [reply src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js:59]: Actually yeah, react-router lets you pass data between screens, totally forgot you could do that!

The only caveat is this data lives in memory, so if you visit this page from a cold start you won't have the data. I'll look into this further and let you know

File: src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js
Comment (vfcheung) [reply src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js:34]: My bad, it's not needed.

File: src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js
Comment (vfcheung) [reply src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js:42]: Good point, this array can just live in a separate file and be imported by whoever needs it.

File: src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js
Comment (vfcheung) [reply src/components/Forum/ForumApprovedPostDetails/ForumPostDetailsList.js:14]: Relative is probably better. Will change.

**Review by vfcheung — COMMENTED** [review]
Thanks for comments! Good suggestions, will make changes.

**Comment by vfcheung** [comment]
Going to hold off on this until the list views get refactored.
Reason: To prevent having to do 3 database fetches, we can store the post's data into react-router's state when we navigate into this page. This means we need to have all the post's data available in the previous page (including the post's category and user's display name). This is not possible right now.

**Review by enmingliu — APPROVED** [review]

**Pull request merged**
